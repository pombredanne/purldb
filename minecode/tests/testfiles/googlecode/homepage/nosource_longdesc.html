

<!DOCTYPE html>
<html>
<head>
 <meta http-equiv="Content-Type" content="text/html; charset=UTF-8" >
 <meta http-equiv="X-UA-Compatible" content="IE=edge,chrome=1" >
 
 <meta name="ROBOTS" content="NOARCHIVE">
 
 <link rel="icon" type="image/vnd.microsoft.icon" href="https://ssl.gstatic.com/codesite/ph/images/phosting.ico">
 
 
 <script type="text/javascript">
 
 
 
 
 var codesite_token = null;
 
 
 var CS_env = {"projectHomeUrl": "/p/bwip-js", "relativeBaseUrl": "", "assetHostPath": "https://ssl.gstatic.com/codesite/ph", "loggedInUserEmail": null, "profileUrl": null, "assetVersionPath": "https://ssl.gstatic.com/codesite/ph/11512667522589661435", "domainName": null, "projectName": "bwip-js", "token": null};
 var _gaq = _gaq || [];
 _gaq.push(
 ['siteTracker._setAccount', 'UA-18071-1'],
 ['siteTracker._trackPageview']);
 
 (function() {
 var ga = document.createElement('script'); ga.type = 'text/javascript'; ga.async = true;
 ga.src = ('https:' == document.location.protocol ? 'https://ssl' : 'http://www') + '.google-analytics.com/ga.js';
 (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(ga);
 })();
 
 </script>
 
 
 <title>
 bwip-js -
 
 
 Barcode Writer in Pure JavaScript - Google Project Hosting
 </title>
 <link type="text/css" rel="stylesheet" href="https://ssl.gstatic.com/codesite/ph/11512667522589661435/css/core.css">
 
 <link type="text/css" rel="stylesheet" href="https://ssl.gstatic.com/codesite/ph/11512667522589661435/css/ph_detail.css" >
 
 
 
 
<!--[if IE]>
 <link type="text/css" rel="stylesheet" href="https://ssl.gstatic.com/codesite/ph/11512667522589661435/css/d_ie.css" >
<![endif]-->
 <style type="text/css">
 .menuIcon.off { background: no-repeat url(https://ssl.gstatic.com/codesite/ph/images/dropdown_sprite.gif) 0 -42px }
 .menuIcon.on { background: no-repeat url(https://ssl.gstatic.com/codesite/ph/images/dropdown_sprite.gif) 0 -28px }
 .menuIcon.down { background: no-repeat url(https://ssl.gstatic.com/codesite/ph/images/dropdown_sprite.gif) 0 0; }
 
 
 .psicon-container {
 min-width:24px;
 }

 
 </style>
</head>
<body class="t1">
<script type="text/javascript">
 window.___gcfg = {lang: 'en'};
 (function() 
 {var po = document.createElement("script");
 po.type = "text/javascript"; po.async = true;po.src = "https://apis.google.com/js/plusone.js";
 var s = document.getElementsByTagName("script")[0];
 s.parentNode.insertBefore(po, s);
 })();
</script>
<div class="headbg">

 <div id="gaia">
 

 <span>
 
 
 <a href="#" id="projects-dropdown" onclick="return false;"><u>My favorites</u> <small>&#9660;</small></a>
 | <a href="https://www.google.com/accounts/ServiceLogin?service=code&amp;ltmpl=phosting&amp;continue=https%3A%2F%2Fcode.google.com%2Fp%2Fbwip-js%2F&amp;followup=https%3A%2F%2Fcode.google.com%2Fp%2Fbwip-js%2F" onclick="_CS_click('/gb/ph/signin');"><u>Sign in</u></a>
 
 </span>

 </div>

 <div class="gbh" style="left: 0pt;"></div>
 <div class="gbh" style="right: 0pt;"></div>
 
 
 <div style="height: 1px"></div>
<!--[if lte IE 7]>
<div style="text-align:center;">
Your version of Internet Explorer is not supported. Try a browser that
contributes to open source, such as <a href="http://www.firefox.com">Firefox</a>,
<a href="http://www.google.com/chrome">Google Chrome</a>, or
<a href="http://code.google.com/chrome/chromeframe/">Google Chrome Frame</a>.
</div>
<![endif]-->



 <table style="padding:0px; margin: 0px 0px 10px 0px; width:100%" cellpadding="0" cellspacing="0"
 itemscope itemtype="http://schema.org/CreativeWork">
 <tr style="height: 58px;">
 
 
 
 <td id="plogo">
 <link itemprop="url" href="/p/bwip-js">
 <a href="/p/bwip-js/">
 
 
 <img src="/p/bwip-js/logo?cct=1361905601"
 alt="Logo" itemprop="image">
 
 </a>
 </td>
 
 <td style="padding-left: 0.5em">
 
 <div id="pname">
 <a href="/p/bwip-js/"><span itemprop="name">bwip-js</span></a>
 </div>
 
 <div id="psum">
 <a id="project_summary_link"
 href="/p/bwip-js/"><span itemprop="description">Barcode Writer in Pure JavaScript</span></a>
 
 </div>
 
 
 </td>
 <td style="white-space:nowrap;text-align:right; vertical-align:bottom;">
 
 <form action="/hosting/search">
 <input size="30" name="q" value="" type="text">
 
 <input type="submit" name="projectsearch" value="Search projects" >
 </form>
 
 </tr>
 </table>

</div>

 
<div id="mt" class="gtb"> 
 <a href="/p/bwip-js/" class="tab active">Project&nbsp;Home</a>
 
 
 
 
 <a href="/p/bwip-js/downloads/list" class="tab ">Downloads</a>
 
 
 
 
 
 
 
 <a href="/p/bwip-js/issues/list"
 class="tab ">Issues</a>
 
 
 
 
 
 
 
 
 
 
 <div class=gtbc></div>
</div>
<table cellspacing="0" cellpadding="0" width="100%" align="center" border="0" class="st">
 <tr>
 
 
 
 
 
 
 
 <td class="subt">
 <div class="st1">
 <div class="isf">
 <span class="inst1">
 <a href="/p/bwip-js/">Summary</a>
 </span>
 
 
 &nbsp;
 <span class="inst3">
 <a href="/p/bwip-js/people/list">People</a>
 </span>
 
 
 </div>
</div>

 </td>
 
 
 <td align="right" valign="top" class="bevel-right"></td>
 </tr>
</table>


<script type="text/javascript">
 var cancelBubble = false;
 function _go(url) { document.location = url; }
</script>
<div id="maincol"
 
>

 





<table width="100%">
 <tr class="pscontent">
 <td class="pscolumnl">
 
 <div class="phead">Project Information</div>
 
 
 
 <ul class="pslist">
 
 <li class="psmeta">
 <a href="/p/bwip-js/feeds">Project feeds</a>
 </li>
 <li class="psgap"/>
 
 
 
 <li class="psmeta"><b>Code license</b></li>
 <li class="psmeta">
 <a href="http://www.opensource.org/licenses/mit-license.php" rel="nofollow">MIT License</a>
 
 </li>
 <li class="psgap"/>
 
 
 
 
 
 
 <li class="psmeta">
 <span id="project_labels">
 <b>Labels</b><br/>
 
 <a class="label"
 href="/hosting/search?q=label:JavaScript">JavaScript</a>, 
 
 <a class="label"
 href="/hosting/search?q=label:Barcode">Barcode</a>, 
 
 <a class="label"
 href="/hosting/search?q=label:Generator">Generator</a>, 
 
 <a class="label"
 href="/hosting/search?q=label:CrossBrowser">CrossBrowser</a>, 
 
 <a class="label"
 href="/hosting/search?q=label:ServerSide">ServerSide</a>, 
 
 <a class="label"
 href="/hosting/search?q=label:CommonJS">CommonJS</a>, 
 
 <a class="label"
 href="/hosting/search?q=label:Node.js">Node.js</a>
 
 </span>
 </li>
 <li class="psgap"/>
 
 
 </ul>
 <div class="psicon">
 <div class="psicon-container goog-inline-block">
 <div style="float:right" class="SPRITE_people-y16 goog-inline-block vt"></div>
 </div>
 <span><b>Members</b></span>
 </div>
 <ul class="pslist">
 
 
 
 

 <a class="userlink"
 href="/u/104115978442467742590/">mwar...@metafloor.com</a
>
 
 
 
 
 <li class="psgap"/>
 
 
 </ul>
 
 
 
 
 </td>
 <td id="wikicontent" class="psdescription">
 <h1><a name="Barcode_Writer_in_Pure_JavaScript"></a>Barcode Writer in Pure JavaScript<a href="#Barcode_Writer_in_Pure_JavaScript" class="section_anchor"></a></h1><p>bwip-js is a translation to native JavaScript of the amazing code provided in <a href="http://code.google.com/p/postscriptbarcode/" rel="nofollow">Barcode Writer in Pure PostScript</a>.  The translated code can run on any browser that natively supports the HTML canvas element or any JavaScript-based server framework that can implement a minimal bitmap graphics interface. </p><p>An in-browser demo is provided that mirrors the functionality of the <a href="http://www.terryburton.co.uk/barcodewriter/generator/" rel="nofollow">web-based demo</a> provided in BWIPP. </p><p>To use the demo, download bwip-js and unpack it.  The download will install to the directory bwip-js.  Then open the file bwip-js/demo.html in your browser and start making barcodes. </p><p>The latest versions - along with many older versions - of Google Chrome, Firefox, Opera and Safari work well with the demo.  Internet Explorer 8 and older will not work as they do not implement the canvas element. </p><p>Version 0.4 of bwip-js (released 2013-02-26) updates to BWIPP version 2013-01-17 and fixes a bug in the node-js module.   </p><h2><a name="Barcontains_source_codes"></a>Barcode Types<a href="#Barcontains_source_codes" class="section_anchor"></a></h2><p>bwip-js implements the same barcode types as BWIPP.  A comprehensive list includes: </p><ul><li>AusPost 4 State Customer Code </li><li>Aztec Code </li><li>BC412 </li><li>Channel Code </li><li>Codablock F </li><li>Code 11 </li><li>Code 128 </li><li>Code 16K </li><li>Code 25 </li><li>Code 39 </li><li>Code 39 Extended </li><li>Code 49 </li><li>Code 93 </li><li>Code 93 Extended </li><li>Code One </li><li>Data Matrix </li><li>Deutsche Post Identcode </li><li>Deutsche Post Leitcode </li><li>EAN-13 </li><li>EAN-13 Composite </li><li>EAN-2 (2 digit addon) </li><li>EAN-5 (5 digit addon) </li><li>EAN-8 </li><li>EAN-8 Composite </li><li>Flattermarken </li><li>GS1 Composite 2D Component </li><li>GS1 Data Matrix </li><li>GS1 DataBar Expanded </li><li>GS1 DataBar Expanded Composite </li><li>GS1 DataBar Expanded Stacked </li><li>GS1 DataBar Expanded Stacked Composite </li><li>GS1 DataBar Limited </li><li>GS1 DataBar Limited Composite </li><li>GS1 DataBar Omnidirectional </li><li>GS1 DataBar Omnidirectional Composite </li><li>GS1 DataBar Stacked </li><li>GS1 DataBar Stacked Composite </li><li>GS1 DataBar Stacked Omnidirectional </li><li>GS1 DataBar Stacked Omnidirectional Composite </li><li>GS1 DataBar Truncated </li><li>GS1 DataBar Truncated Composite </li><li>GS1-128 </li><li>GS1-128 Composite </li><li>GS1-14 </li><li>HIBC Codablock F </li><li>HIBC Code 128 </li><li>HIBC Code 39 </li><li>HIBC Data Matrix </li><li>HIBC MicroPDF417 </li><li>HIBC PDF417 </li><li>HIBC QR Code </li><li>ISBN </li><li>ISMN </li><li>ISSN </li><li>ITF-14 </li><li>Interleaved 2 of 5 (ITF) </li><li>Italian PharmaCode </li><li>Japan Post 4 State Customer Code </li><li>MSI Modified Plessey </li><li>MaxiCode </li><li>MicroPDF417 </li><li>Miscellaneous symbols </li><li>PDF417 </li><li>Pharmaceutical Binary Code </li><li>Pharmazentralnummer (PZN) </li><li>Plessey UK </li><li>PosiCode </li><li>QR Code </li><li>Rationalized Codabar </li><li>Raw DAFT succession for custom 4 state symbologies </li><li>Raw bar space succession for custom symbologies  </li><li>Royal Dutch TPG Post KIX 4-State Barcode </li><li>Royal Mail 4 State Customer Code (RM4SCC) </li><li>SSCC-18 </li><li>Telepen </li><li>Two-track Pharmacode </li><li>UPC-A </li><li>UPC-A Composite </li><li>UPC-E </li><li>UPC-E Composite </li><li>United States Postal Service Intelligent Mail </li><li>United States Postal Service PLANET </li><li>United States Postal Service POSTNET </li></ul><h2><a name="Features"></a>Features<a href="#Features" class="section_anchor"></a></h2><p>bwip-js is almost feature complete.  At present, it lacks the following functionality that exists in BWIPP: <ul><li>Minimal font support.  A public-domain, unencumbered version of the OCR-B typeface has been used to create a series of bitmapped fonts.  These fonts were sized to be compatible with the default 10pt and 12pt font sizes used by BWIPP. </li><li>Only orthogonal scaling is supported.  This restriction is due to the use of bitmapped fonts. </li><li>Rotations of 0, 90, 180, and 270 degrees are supported only in the HTML canvas demo.  There are no plans to implement rotation in the PostScript graphics emulation layer. </li><li>Inkspread is currently hard-coded to 0 in the demo.  Setting it to other values may create non-compliant barcodes. </li><li>The DataMatrix barcodes do not match the size output by BWIPP (bwip-js barcodes are smaller).  You must use the scale setting to compensate. </li></ul></p><p>bwip-js does <i>not</i> provide a general purpose PostScript emulation layer.  If you came here hoping to find one, you are out of luck.  The PostScript emulation uses a translator that converts all non-graphics code to equivalent JavaScript, and converts all graphics operations to calls into a platform-neutral graphics context.  It is up to the host environment to implement the bitmap interface that is exposed by the graphics context.  An example implementation is provided in the demo.  </p><p>The PostScript to JavaScript cross-compiler is not included as part of this project; only the translated code is provided.  The cross-compiler is still in its infancy and implements a one-to-one, unoptimized translation of the stack-based PostScript operators.  There is a lot of opportunity to perform peephole optimization and other transforms that convert the code into more efficient JavaScript. </p><p>Although the code is not as efficient as it could be, it is still surprisingly fast.  Only with math-intensive symbols (e.g. QRCode with a high level of error correction) is there any appreciable lag. </p><h2><a name="Installation"></a>Installation<a href="#Installation" class="section_anchor"></a></h2><p>Go to <a href="http://code.google.com/p/bwip-js/downloads/list" rel="nofollow">Downloads</a> and save the most recent bwip-js package to your computer. </p><p>Unzip the download package.  bwip-js will be unpacked into the following directory structure: </p><pre class="prettyprint">bwip-js/
	bwip.js			# Main bwip-js module
	demo.html		# The in-browser demo
	node-demo		# An example node-js HTTP server application
	node-bwipjs		# A node-js module that implements bwip-js
	bwipp/			# The cross-compiled BWIPP encoders and renderers
	fonts/			# JavaScript-encoded font files
	lib/			# Utilities required by the demo
</pre><h2><a name="Usage"></a>Usage<a href="#Usage" class="section_anchor"></a></h2><p>bwip-js was designed to run in both client-side and server-side JavaScript environments.  For server-side environments that do not provide a graphics API, there are freely available pure-JavaScript implementations of image file formats like PNG and BMP.  A quick search of the web turns up several possibilities that can be adapted for server-side usage.  See node-bwipjs for an example implementation. </p><p>Using bwip-js in your code involves the following steps: <ul><li>Load the main &quot;bwip.js&quot; module. </li><li>Implement two host-specific interfaces. </li><li>Create a BWIPJS instance and make a barcode. </li></ul></p><p>Loading &quot;bwip.js&quot; into your JavaScript environment tends to be very platform specific.  If you are working with a CommonJS architecture, then you will likely need to implement a runtime JavaScript loader.  The node-bwipjs file gives an example implementation for node-js:  </p><pre class="prettyprint">var fs = require(&#x27;fs&#x27;);
var vm = require(&#x27;vm&#x27;);

function load(path) {
    var text = fs.readFileSync(path);
    if (!text)
        throw new Error(path + &quot;: could not read file&quot;);
    vm.runInThisContext(text, path);
}

// Load the primary bwip-js script
load(&#x27;bwip.js&#x27;);</pre><p>After bwip.js is loaded and the BWIPJS global is accessible, you must implement two interfaces: <ul><li>Load and execute bwip-js files on demand.  Many of the encoders rely on additional/secondary encoders.  These dependencies are implemented in the PostScript cross-compile as on-demand loading.  Your host implementation must be able to load and execute bwip-js files after the primary encoder has been invoked.  Or you can preload all of the encoders in one shot (that&#x27;s the strategy the demo takes), but that is a pretty heavy hit every time you generate a barcode. </li><li>Provide a bitmap graphics interface.  One gotcha here is that you won&#x27;t know the size of the bitmap until after the encoders have run.  To make it more interesting, you will routinely see negative values for the x,y coordinates.  Therefore, creating a barcode image is always a two step process.  First run the encoder/renderer and record each pixel&#x27;s coordinates, along with the min/max values for x and y.  Once the encoder has returned, you can determine the bitmap dimensions and render the final image. </li><li>The other gotcha is that typical graphics environments (HTML canvas, PHP GD, etc.) set their origin (0,0) in the upper left-hand corner.  PostScript uses a page-up orientation and the origin is in the lower left-hand corner.  Since bwip-js implements a direct mapping to the PostScript graphics primitives, the coordinates your bitmap implementation will see have an origin set in the lower left-hand corner.  See the files lib/canvas.js and node-bwipjs for bitmap implementations that convert between the two origin conventions. </li></ul></p><p>Below are the function and object prototypes for the interfaces to implement: </p><pre class="prettyprint">// Demand-load encoders, renderers, and fonts.  
// This function is installed as a &quot;class static&quot; method on the bwip-js constructor.
BWIPJS.load = function(path) {
	// path is relative to the bwip-js root directory.  For example, a call to
	// load the one-dimensional barcode renderer will pass the path:
	//		&quot;bwipp/renlinear.js&quot;
	// This implementation must translate the relative path to whatever is
	// required of the server framework and directory structure.
	// bwip-js tracks what has already been loaded (regardless of whether a
	// module was preloaded or demand-loaded).
	// So this interface will only be called when needed.
	...
}

// The bitmap interface.  bwip-js will only call the color() and set() 
// methods during encoding/drawing.  A third method is needed by the 
// host code to render the actual barcode image.
// Note that you can name this object class anything you like.  The
// host code is the only one that references it by name.
function Bitmap() {
	this.color = function(r, g, b) {
		// r,g,b will be integer values between 0 and 255
		// Save as the current color
		...
	};
	this.set = function(x, y) {
		// x,y will be floating point values.  Convert to int.
		x = Math.floor(x);
		y = Math.floor(y);
		
		// Save the coordinates and current color
		...
		// Track the min/max values of x and y
		...
	};
	this.render = function( ... ) {
		// This is your code to render the final bitmap image.
		// Width of the bitmap is:  max_x - min_x + 1
		// Height of the bitmap is: max_y - min_y + 1
		// You will need to offset the pixel coordinates by min_x and min_y.
		// And invert the y-axis if the graphics format you are using has its
		// origin in the upper-left corner.
		...
	};
}</pre><p>Once the host framework is implemented, it&#x27;s time to invoke an encoder.  You must remember that this is a PostScript emulation, so the calling conventions have a PostScript feel to them.   </p><p>BWIPP (the PostScript library) uses a two-parameter calling convention.  The first parameter is always the barcode text to be encoded.  The second parameter is the options that govern how the barcode is generated.  This second parameter can be either a string or a PostScript dictionary object.  This author recommends using a dictionary object for the options, as they provide more flexibility when working with options values.  </p><p>One of the standard options available for linear/one-dimensional barcodes is the alttext option.  This option allows overriding what is printed as the human readable component of the barcode.  If the options are specified as a string, no spaces can occur in the alttext value.  This restriction does not exist if a dictionary object is used. </p><p>To invoke a barcode encoder/renderer, you do the following steps: <ul><li>Install the host bitmap interface. </li><li>Create the dictionary object for the options parameter. </li><li>Add the necessary options to the object. </li><li>Push the barcode text onto the PostScript operand stack. </li><li>Push the options object onto the PostScript operand stack. </li><li>Invoke the encoder.  bwip-js will automatically demand load the module, if needed. </li></ul>  This next example calls the code128 encoder using the &#x27;parsefnc&#x27; option to specify the FNC1 character as part of the barcode text, and provides an alternate, human readable text. </p><pre class="prettyprint">// Create a barcode writer instance
var bw = new BWIPJS;

// Create the bitmap interface and pass to the emulator
bw.bitmap(new Bitmap);

// Set the scaling factor
bw.scale(2, 2);

// Create a dictionary object and set the options
var opts = {};
opts.parsefnc    = bw.value(true);
opts.includetext = bw.value(true);
opts.alttext     = bw.value(&quot;(00)1234567890&quot;);

// Push the barcode text and options onto the operand stack
bw.push(&quot;^FNC1001234567890&quot;);
bw.push(opts);

// Invoke the encoder and render the barcode
bw.call(&#x27;code128&#x27;);
bw.bitmap().render();</pre><h2><a name="Node.js"></a>Node.js<a href="#Node.js" class="section_anchor"></a></h2><p><a href="http://nodejs.org" rel="nofollow">Node.js</a> has gained a lot of traction as a server framework but due to its rather unique architecture, integrating bwip-js for use in Node is not a straightforward task.  Included in the bwip-js download (starting with version 0.2) is a fully functional Node module that implements a standard HTTP request handler.  The included example server application demonstrates its usage: </p><pre class="prettyprint">var http = require(&#x27;http&#x27;);
var bwipjs = require(&#x27;./node-bwipjs&#x27;);

http.createServer(function(req, res) {
    // Send all requests to bwip-js.
    // In a real http server, you will want to route requests a bit more
    // intelligently...
    bwipjs(req, res);
}).listen(3030);</pre><p>To use the demo Node server, perform the following steps: <ul><li>Install node.js from the <a href="http://nodejs.org" rel="nofollow">Download</a> link. </li><li>Download and unzip the latest bwip-js <a href="http://code.google.com/p/bwip-js/downloads/list" rel="nofollow">version</a>. </li><li>Open a terminal window/command prompt and &#x27;cd&#x27; to the bwip-js directory. </li><li>Run the demo server: </li></p><pre class="prettyprint">  $ node node-demo</pre></ul><p>Once the server is running on your machine, the following link should display a Code 128 bar code in your browser: </p><p><a href="http://127.0.0.1:3030/?bcid=code128&amp;text=^FNC1011234567890&amp;scale=4&amp;rotate=L&amp;parsefnc&amp;alttext=(01)01234567890" rel="nofollow">http://127.0.0.1:3030/?bcid=code128&amp;text=^FNC1011234567890&amp;scale=4&amp;rotate=L&amp;parsefnc&amp;alttext=(01)01234567890</a> </p><p>See the notes in node-bwipjs for details on how to structure the URL for use with the request handler. </p><p>The node-bwipjs module provides an example of how to integrate bwip-js into node.js.  It is not production ready.  In particular, the PNG image support is extremely simplistic and will waste a lot of server bandwidth.  256 color depth is used (most bar codes need only two colors) and DEFLATE compression is not implemented.  A native Node implementation of the PNG library with DEFLATE support should greatly reduce image size and CPU utilization. </p><h2><a name="Emulator_Notes"></a>Emulator Notes<a href="#Emulator_Notes" class="section_anchor"></a></h2><p>The emulation uses, as much as possible, native JavaScript data types.  Specifically, the following are used by the emulation: <ul><li>Boolean values true and false </li><li>Null value </li><li>Numeric values (integer and float) </li><li>Object (PostScript dictionary object) </li></ul></p><p>You will note that two of the more common JavaScript data types, strings and arrays, are not on the list.  PostScript implements some very powerful semantics with strings and arrays, namely live views into each data type.  For example, you can have a base array plus other references to the array that are offset views into the array elements.  Each view has its own concept of zero-index and length, but these values may or may not match the base array.  Updates to one view causes the base array to be updated and are visible to all other views of the array. </p><p>Similar live-view semantics exist for PostScript strings.  Additionally, strings behave more like arrays of character codes rather than JavaScript&#x27;s usual string semantics.  (PostScript strings actually behave like Uint8 Typed Arrays; if Typed Arrays ever become common place, we can gain some performance by converting the emulation to use it for strings.  The same cannot be said for arrays, since PostScript arrays can store arbitrary data types, similar to JavaScript.) </p>
 </td>
 </tr>
</table>
<script src="https://ssl.gstatic.com/codesite/ph/11512667522589661435/js/prettify/prettify_core_compiled.js"></script>
<script type="text/javascript">
 prettyPrint();
</script>

 
 
 
 <script type="text/javascript" src="https://ssl.gstatic.com/codesite/ph/11512667522589661435/js/ph_core.js"></script>
 
 
 
 
</div> 

<div id="footer" dir="ltr">
 <div class="text">
 <a href="/projecthosting/terms.html">Terms</a> -
 <a href="http://www.google.com/privacy.html">Privacy</a> -
 <a href="/p/support/">Project Hosting Help</a>
 </div>
</div>
 <div class="hostedBy" style="margin-top: -20px;">
 <span style="vertical-align: top;">Powered by <a href="http://code.google.com/projecthosting/">Google Project Hosting</a></span>
 </div>

 
 


 
 </body>
</html>

