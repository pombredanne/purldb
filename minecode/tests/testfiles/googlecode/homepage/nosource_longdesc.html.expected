{'content_license': None,
 'content_license_url': None,
 'description': u'Barcode Writer in Pure JavaScript',
 'downloads_url': u'http://code.google.com/p/bwip-js/downloads/list',
 'external_links': {},
 'feeds_url': u'http://code.google.com/p/bwip-js/feeds',
 'home_url': u'http://code.google.com/p/bwip-js/',
 'id': u'bwip-js',
 'issues_url': u'http://code.google.com/p/bwip-js/issues/list',
 'keywords': [],
 'labels': [u'Barcode',
            u'CommonJS',
            u'CrossBrowser',
            u'Generator',
            u'JavaScript',
            u'Node.js',
            u'ServerSide'],
 'license': u'MIT License',
 'license_url': u'http://www.opensource.org/licenses/mit-license.php',
 'name': u'bwip-js',
 'peoples_url': u'http://code.google.com/p/bwip-js/people/list',
 'vcs_url': None,
 'wiki_url': None}